airflow/providers/teradata/operators/teradata.py

```diff
@@ -62,3 +66,38 @@ def __init__(
             }
         super().__init__(**kwargs)
         self.conn_id = conn_id
+
+
+class TeradataStoredProcedureOperator(BaseOperator):
+    """
+    Executes stored procedure in a specific Teradata database.
+
+    :param procedure: name of stored procedure to call (templated)
+    :param conn_id: The :ref:`Teradata connection id <howto/connection:teradata>`
+        reference to a specific Teradata database.
+    :param parameters: (optional, templated) the parameters provided in the call
+
+    """
+
+    template_fields: Sequence[str] = (
+        "procedure",
+        "parameters",
+    )
+    ui_color = "#ededed"
+
+    def __init__(
+        self,
+        *,
+        procedure: str,
+        conn_id: str = TeradataHook.default_conn_name,
```

Comment:
I thought better to use `teradata_conn_id` instead rather than generic `conn_id`

Reply:
Modified conn_id to teradata_conn_id. It requires to change system tests. Changed system tests as per this modification.